pages/upload.js

```diff
@@ -26,37 +31,46 @@ function Upload(props) {
       .insert([
         {
           mood: inputs.mood,
+          user_id: sessionId.user.id,
           alt: inputs.alt,
           caption: inputs.caption,
           public: inputs.public,
           prompt: inputs.prompt,
           img: inputs.img,
+          email: sessionId.user.email,
         },
       ])
       .then((data) => {
         return data;
       });
 
-    if (!moodArray.includes(inputs.mood)) {
+    if (!everymoodarr.includes(inputs.mood)) {
+      console.log("inserting");
+      const { data, error } = await supabase.from("moods").insert([
+        {
+          mood: inputs.mood,
+          count: 1,
+        },
+      ]);
+    } else {
+      console.log("else statement");
       const { data, error } = await supabase
-        .from("moods")
-        .insert([
-          {
-            mood: inputs.mood,
-            count: 1,
-          },
-        ])
+        .rpc("increment", {
```

Comment:
Filtering the moods will definitely improve the user experience. I like how you're listening out to see if the mood exists and keeping count